### ISSUE TITLE
Ip cam not streaming

### ISSUE THREAD

**Issue by koeniebert** [swttt/HomeyDash#42]
**What OS and browser are you using?**

Google Chrome on Macbook

**What did you try to achieve?**

Adding ip cam. Screen is ok but it is a fixed image. No stream.

**What was the result?**

Screen is ok but it is a fixed image. No stream.

**What did you expect as result?**

A streaming video.

**Any other remarks**

...

**Comment by jghaanstra** [comment]
@koeniebert what do you see when you access the stream URL directly in the browser?

**Comment by koeniebert** [comment]
Also a steady image.
It's a little strange because the stream works at the "old" Homeydash.

**Comment by jghaanstra** [comment]
Could it be there is a limit on simultaneous connections to the camera causing it to display a static image when it's over it's limit?

And does the stream work in the "old" HomeyDash when viewing this "old" dashboard on the same machine as where you are now viewing the "new" HomeyDash?

**Comment by koeniebert** [comment]
Yes working on the "old" HomeyDash on the same platform on the same machine. Could it be that the "old" HomeyDash is refreshing every second?

**Comment by koeniebert** [comment]
Could it be the problem that it's a JPEG stream?

**Comment by jghaanstra** [comment]
The stream **should** be MJPEG, that's needed for this widget to work.

It seems the old HomeyDash is doing new http requests every second. Because the new HomeyDash is client side javascript this works differently. When you access the URL directly through the browser and open the developer console (F12), then select the network tab and refresh the page, what do you see?

But since accessing the stream directly also gives you a static image this is not actually an issue with HomeyDash or the ipwebcam widget.

**Comment by koeniebert** [comment]
Ok, so it's not possible to get an stream on HomeyDash with this ipcam because it uses a JPEG?

**Comment by swttt** [comment]
This should be possible by extending the widget to refresh every second indeed. Could be a setting or something. Will look into it when I feel good enough... damn virus

**Comment by jghaanstra** [comment]
@koeniebert if your camera has a snapshot url (which returns a jpeg) at could very well also have a video url (which will most likely return a mjpeg stream). Have you looked into this?

And like swttt says, it's possible but hammering I'm not sure if it's desired. Polling a snapshot URL every second could generate a lot more traffic if these snapshots are of higher quality than the mjpeg stream.

**Comment by koeniebert** [comment]
I will look into it but i think there is not a mjpeg stream.
@swttt get well soon for you!

**Comment by koeniebert** [comment]
Did looking into it, i have a live stream with a rtsp:// path. Working on VLC player. Is it possible to implement this in the widget?

**Comment by jghaanstra** [comment]
No, rtsp:// is not a webbased protocol. It would require some middleware to convert the stream in something suitable for the web but that is beyond the scope of a simple mjpeg streaming widget.

**Comment by swttt** [comment]
Not sure if and how it works on Android/iOS but might be an option to look into VLC.

No idea if it Could work, but just popped into my head

**Comment by koeniebert** [comment]
Hoi Guys,
Maybe you can help me out. Trying to fix this for my ip camera. I set up a stream from rtsp to http on my mac using VLC player. The stream works, but how can i acces the stream? Just going to http://<ip-adres>:8080 is not working.

Maybe you can help me out.

**Comment by jghaanstra** [comment]
If VLC is running on the same machine as HomeyDash and VLC has converted the rtps stream to a http stream you should be able to access the stream with http://<<your machine ip>>:<<port of http stream>> or http://127.0.0.1:<<port of http stream>>.

**Comment by koeniebert** [comment]
No sorry, not working.
Any other ideas before i go to buy a new camera?
